packages/neuron-wallet/src/block-sync-renderer/sync/light-synchronizer.ts

```diff
@@ -198,32 +211,30 @@ export default class LightSynchronizer extends Synchronizer {
         }))
       })
       .flat()
-    const walletMinBlockNumber = await SyncProgressService.getWalletMinLocalSavedBlockNumber()
-    const wallets = await WalletService.getInstance().getAll()
-    const walletStartBlockMap = wallets.reduce<Record<string, string | undefined>>(
-      (pre, cur) => ({ ...pre, [cur.id]: cur.startBlockNumber }),
-      {}
-    )
+      .filter(v => {
+        return (
+          !currentWalletId ||
+          v.walletId === currentWalletId ||
+          walletMinBlockNumber[v.walletId] > walletMinBlockNumber[currentWalletId] - thresholdBlockNumberInDiffWallet
+        )
+      })
     const otherTypeSyncBlockNumber = await SyncProgressService.getOtherTypeSyncBlockNumber()
     const addScripts = [
       ...allScripts
-        .filter(
-          v =>
-            !existSyncscripts[scriptToHash(v.script)] ||
-            +existSyncscripts[scriptToHash(v.script)].blockNumber < +(walletStartBlockMap[v.walletId] ?? 0)
-        )
-        .map(v => {
-          const blockNumber = Math.max(
-            parseInt(walletStartBlockMap[v.walletId] ?? '0x0'),
-            walletMinBlockNumber?.[v.walletId] ?? 0
+        .filter(v => {
+          return (
+            !existSyncScripts[scriptToHash(v.script)] ||
+            +existSyncScripts[scriptToHash(v.script)].blockNumber < walletMinBlockNumber[v.walletId]
```

Comment:
`scriptToHash(v.script)` can be assigned to a variable to avoid hashing it again, or simply

```js
+(exsitSyncScript[scriptToHash(v.script)]?.blockkNumber ?? 0) < walletMinBlockNumber[v.walletId]
```